packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1033,10 +1042,14 @@ actual object RealmInterop {
     ): RealmQueryPointer {
         memScoped {
             val count = args.size
-            val cArgs = allocArray<realm_value_t>(count)
+            val cArgs = allocArray<realm_query_arg_t>(count)
```

Comment:
Looks like a code block that are used a number of times. Should probably be abstracted into a method and reused instead of copying it.

Reply:
I've extracted it to a separate function and added a comment to clarify the usage of `realm_query_arg_t` as it is a bit quirky.